src/main/java/com/metamx/common/io/smoosh/SmooshedFileMapper.java

```diff
@@ -131,10 +132,20 @@ public ByteBuffer mapFile(String name) throws IOException
   }
 
   @Override
-  public void close() throws IOException
+  public void close()
   {
+    Throwable thrown = null;
     for (MappedByteBuffer mappedByteBuffer : buffersList) {
-      ByteBufferUtils.unmap(mappedByteBuffer);
+      try {
+        ByteBufferUtils.unmap(mappedByteBuffer);
+      } catch (Throwable t) {
+        if (thrown == null) {
+          thrown = t;
+        } else {
+          thrown.addSuppressed(t);
+        }
+      }
     }
+    Throwables.propagateIfPossible(thrown);
```

Comment:
`IOException` was not possible here, so I just removed unneeded `throws IOException`. Only `RuntimeException` or `Error` are possible. So `Throwables.propagateIfPossible(thrown);` is just a way to rethrow, without javac complaining.